themes/material/src/classes/tooltips_cls.rs

```diff
@@ -0,0 +1,52 @@
+use ribir_core::prelude::*;
+
+class_names! {
+  TOOLTIPS_BASE,
+}
+
+pub(super) fn init(classes: &mut Classes) {
+  classes.insert(TOOLTIPS_BASE, |w| {
+    fn_widget! {
+        let w = FatObj::new(w);
+        @BoxDecoration {
+          background: Palette::of(BuildCtx::get()).inverse_surface(),
+          margin: EdgeInsets::only_bottom(4.),
+          border_radius: Radius::all(4.),
+          @ $w {
+              margin: EdgeInsets::new(4., 8., 4., 8.),
+              foreground: Palette::of(BuildCtx::get()).inverse_on_surface(),
+              v_align: VAlign::Center,
+              h_align: HAlign::Center,
+          }
+        }
+    }
+    .into_widget()
+  });
+
+  classes.insert(TOOLTIPS, |w| {
+    fn_widget! {
+      let w = FatObj::new(w);
+      let mut w = @ $w {
+        class: TOOLTIPS_BASE,
+      };
+      $w.write().keep_alive = true;
+
+      let animate = part_writer!(&mut w.opacity)
+          .transition(transitions::LINEAR.of(BuildCtx::get()));
+      @ $w {
```

Comment:
It appears that the keep-alive control can be simplified as follows:

```rust
@ $w {
  keep_alive: pipe!(!$aniamte.is_running() && $w.opacitfy == 0.)
}
```

Reply:
If this solution works, I believe we may no longer require a base class; it seems straightforward enough.